src/etools/libraries/locations/tasks.py

```
@@ -225,21 +233,29 @@ def cleanup_obsolete_locations(self, carto_table_pk):
     remapped_pcodes += [remap_row['new_pcode'] for remap_row in remapped_pcode_pairs]
     deleteable_pcodes = set(database_pcodes) - (set(new_carto_pcodes) | set(remapped_pcodes))
 
+    revalidated_deleteable_pcodes = []
     for deleteable_pcode in deleteable_pcodes:
         try:
-            deleteable_location = Location.objects.all_locations().get(p_code=deleteable_pcode)
+            deleteable_location = Location.objects.all_locations().get(
+                p_code=deleteable_pcode,
+                gateway=carto_table.location_type,
```

can we let this fail in case there are multiple duplicated p_codes in the same country?
We've talked to Jan and we understand this is a possibility currently but would rather fix it than accommodate for it.